controlplane/internal/controller/openshiftassistedcontrolplane_controller_test.go

```diff
@@ -31,13 +31,21 @@ import (
 	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
-	controlplanev1alpha1 "github.com/openshift-assisted/cluster-api-agent/controlplane/api/v1alpha1"
+	controlplanev1alpha2 "github.com/openshift-assisted/cluster-api-agent/controlplane/api/v1alpha2"
 	testutils "github.com/openshift-assisted/cluster-api-agent/test/utils"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+type mockOpenShiftVersioner struct {
+}
+
+func (m *mockOpenShiftVersioner) GetK8sVersionFromReleaseImage(ctx context.Context, releaseImage string, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) (*string, error) {
+	version := "1.30.0"
```

Comment:
Are you planning on making this a todo to get the version from the release image?

Reply:
This is the mock/dummy in the test, we are already  - I can add an assertion making sure status.version is 1.30.0
the real thing is happening here https://github.com/openshift-assisted/cluster-api-agent/blob/distribution-version/controlplane/internal/controller/openshiftassistedcontrolplane_controller.go#L172